src/serac/infrastructure/profiling.hpp

```diff
@@ -118,6 +141,36 @@ void setCaliperMetadata(const std::string& name, double data);
   @overload
 */
 void setCaliperMetadata(const std::string& name, unsigned int data);
+
+/**
+ * @brief Caliper method for marking the start of a profiling region
+ *
+ * @param[in] name The tag to associate with the region.
+ */
+void startCaliperRegion(const char* name);
+
+/*!
+  @overload
+*/
+void startCaliperRegion(const std::string& name);
+
+/**
+ * @brief Caliper methods for marking the end of a region
+ *
+ * @param[in] name The tag to associate with the region.
+ */
+void endCaliperRegion(const char* name);
+
+/*!
+  @overload
+*/
+void endCaliperRegion(const std::string& name);
+
+template <typename T>
+struct expr_t {
+  using expr_type = T;
+};
+
```

Comment:
Minor: Could we just use the implementation of [std::type_identity](https://en.cppreference.com/w/cpp/types/type_identity) here so it's a drop-in replacement when C++20 is available?